File: src/main/java/io/github/tanguygab/conditionalactions/actions/types/other/DelayAction.java
package io.github.tanguygab.conditionalactions.actions.types.other;

import io.github.tanguygab.conditionalactions.actions.types.Action;
import org.bukkit.OfflinePlayer;

public class DelayAction extends Action {

    public DelayAction() {
        super("^(?i)delay:( )?");
    }

    @Override
    public String getSuggestion() {
        return "delay: <ms>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        match = parsePlaceholders(player,match);
        try {Thread.sleep(parseInt(match,1000));}
        catch (InterruptedException e) {throw new RuntimeException(e);}
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/other/DelayAction.kt
package io.github.tanguygab.conditionalactions.actions.types.other

import io.github.tanguygab.conditionalactions.actions.Action
import org.bukkit.OfflinePlayer

class DelayAction : Action("^(?i)delay:( )?".toRegex()) {
    override fun getSuggestion() = "delay: <ms>"

    override fun execute(player: OfflinePlayer?, match: String) {
        val match = parsePlaceholders(player, match)
        Thread.sleep(parseInt(match, 1000).toLong())
    }
}
